lib/metrics/statsd.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const normalizeName = require('./utils').normalizeName;
+const formatLabels = require('./utils').formatLabels;
+
+// StatsD Metrics implementation
+class StatsDMetric {
+    constructor(options, client) {
+        this.options = options;
+        this.client = client;
+    }
+
+    increment(amount, labels) {
+        labels = formatLabels(this.options, labels);
+        this.client.increment(normalizeName(labels.join('.')), amount);
+    }
+
+    decrement(amount, labels) {
+        labels = formatLabels(this.options, labels);
+        this.client.decrement(normalizeName(labels.join('.')), amount);
+    }
+
+    observe(value, labels) {
+        labels = formatLabels(this.options, labels);
+        this.client.gauge(normalizeName(labels.join('.')), value);
+    }
+
+    gauge(amount, labels) {
+        labels = formatLabels(this.options, labels);
+        this.client.gauge(normalizeName(labels.join('.')), amount);
+    }
+
+    set(value, labels) {
+        labels = formatLabels(this.options, labels);
+        this.client.gauge(normalizeName(labels.join('.')), value);
+    }
+
+    timing(value, labels) {
+        labels = formatLabels(this.options, labels);
+        this.client.timing(normalizeName(labels.join('.')), value);
```

Comment:
To avoid code duplication we can probably make a method `_call(methodName, value, labels)` and call it from all the methods.